packages/twenty-server/src/engine/api/graphql/workspace-query-builder/factories/record-position-query.factory.ts

```diff
@@ -1,54 +1,122 @@
 import { Injectable } from '@nestjs/common';
 
 export enum RecordPositionQueryType {
-  GET = 'GET',
-  UPDATE = 'UPDATE',
+  FIND_FIRST_RECORD = 'FIND_FIRST_RECORD',
+  FIND_LAST_RECORD = 'FIND_LAST_RECORD',
+  FIND_BY_POSITION = 'FIND_BY_POSITION',
+  UPDATE_POSITION = 'UPDATE_POSITION',
 }
 
+type FindByPositionQueryArgs = {
+  positionValue: number;
+  recordPositionQueryType: RecordPositionQueryType.FIND_BY_POSITION;
+};
+
+type FindFirstRecordQueryArgs = {
+  recordPositionQueryType: RecordPositionQueryType.FIND_FIRST_RECORD;
+};
+
+type FindLastRecordQueryArgs = {
+  recordPositionQueryType: RecordPositionQueryType.FIND_LAST_RECORD;
+};
+
+type UpdatePositionQueryArgs = {
+  recordId: string;
+  positionValue: number;
+  recordPositionQueryType: RecordPositionQueryType.UPDATE_POSITION;
+};
+
+type RecordPositionQuery = string;
+
+type RecordPositionQueryParams = any[];
+
+export type RecordPositionQueryArgs =
+  | FindByPositionQueryArgs
+  | FindFirstRecordQueryArgs
+  | FindLastRecordQueryArgs
+  | UpdatePositionQueryArgs;
+
 @Injectable()
 export class RecordPositionQueryFactory {
-  async create(
-    recordPositionQueryType: RecordPositionQueryType,
-    positionValue: 'first' | 'last' | number,
+  create(
+    recordPositionQueryArgs: RecordPositionQueryArgs,
     objectMetadata: { isCustom: boolean; nameSingular: string },
     dataSourceSchema: string,
-  ): Promise<string> {
+  ): [RecordPositionQuery, RecordPositionQueryParams] {
     const name =
```

Comment:
I know it's not something you modified here, but this also feel like it could be in an util